tasks/managed/prepare-and-finalize-artifacts/prepare-and-finalize-artifacts.yaml

```
@@ -78,17 +79,36 @@ spec:
       script: |
         #!/usr/bin/env bash
         set -euxo pipefail
-
         if [ -n "$(params.rpmDataArtifact)" ]; then
-          ARTIFACT_TO_PULL_WITH_PREFIX="$(params.rpmDataArtifact)"
+          ARTIFACT_TO_USE="$(params.rpmDataArtifact)"
```

Maybe it's worth adding a validation to ensure at least one artifact is provided.
something like:
```suggestion
          ARTIFACT_TO_USE="$(params.rpmDataArtifact)"
        elif [ -n "$(params.baseDataArtifact)" ]; then
          ARTIFACT_TO_USE="$(params.baseDataArtifact)"
        else
          echo "Error: No artifact provided. Either baseDataArtifact or rpmDataArtifact must be specified"
          exit 1
        fi
```

---

I added it, thanks for the suggestion.